testdata/go/v3/memcached-operator/test/e2e/e2e_test.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	"github.com/example/memcached-operator/test/utils"
+)
+
+// namespace store the ns where the Operator and Operand will be executed
+const namespace = "memcached-operator-system"
+
+var _ = Describe("memcached", func() {
+
+	Context("ensure that Operator and Operand(s) can run in restricted namespaces", func() {
+		BeforeEach(func() {
+			By("installing prometheus operator")
+			Expect(utils.InstallPrometheusOperator()).To(Succeed())
+
+			By("installing the cert-manager")
+			Expect(utils.InstallCertManager()).To(Succeed())
```

Comment:
Are these steps necessary? It doesn't seem like the test is doing anything regarding metrics. I am also not sure if anything that is being done in the test would require cert-manager.

Reply:
A few questions:
1. For a beginner in operator-SDK do we need e2e tests with webhooks? IMO webhooks are an advanced topic. Can't we disable the webhooks and run. 
2. Also, `testdata/` is a the place where we intend that user looks for a sample operator. If we are adding these e2e tests, I think we should mention this somewhere in our docs too explaining what these e2e tests are intended for. Especially if we decide to have webhooks, a new user wouldn't be aware of these pre-requisites at all.

Reply:
@varshaprasad96,
 
The samples are scaffolded with webhooks and with all options enable. So it is scaffold with cert-manager enable. We are not changing the sample here at all. We are only adding an e2e test as an example to show how the users can do that. 

(ihmo) The samples are examples are useful for:
- Help the users know how they can be built their solutions ( example )
- For the maintainer to check how the changes made in the tool affect the scaffolds, in this way also in the review and for we test the commands

The motivation here would be to link in the docs to show the author how they can test their solutions. I think we should have docs about that but it seems a gap for SDK and KB indeed. But we do things on top of the samples because they are our examples.

Reply:
I added comments that clarify.